auth/store.go

```diff
@@ -33,16 +33,19 @@ var (
 type AuthStore interface {
 	// AuthEnable() turns on the authentication feature
 	AuthEnable()
+
+	// UpdateBackend updates authStore.be during EtcdServer.applySnapshot()
+	UpdateBackend(b backend.Backend)
```

Comment:
probably Recover? Some recovering needs to be done, like reading the auth info from new backend to rebuild the in memory state.